src/mpi-base.jl

```diff
@@ -525,6 +525,36 @@ function Reduce{T}(object::T, op::Op, root::Integer, comm::Comm)
     isroot ? recvbuf[1] : nothing
 end
 
+function Allreduce{T}(sendbuf::MPIBuffertype{T}, recvbuf::MPIBuffertype{T}, 
```

Comment:
The naming convention is that a mixed-case name is a low-level routine (see `Recv!`) that directly wraps the respective MPI function, whereas a higher level routine (see `recv`) is more automatic, i.e. allocating buffers automatically.

I would use the same naming convention here.